clients/src/main/java/org/apache/kafka/common/security/auth/SaslExtensions.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.security.auth;
+
+import org.apache.kafka.common.utils.Utils;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class SaslExtensions {
+    protected Map<String, String> extensionMap;
+    protected String separator;
+
+    protected SaslExtensions() {
+        separator = ",";
+        extensionMap = new HashMap<>();
+    }
+
+    public SaslExtensions(String extensions, String separator) {
```

Comment:
Should this constructor accept the same parameters as Utils.parseMap()?

Reply:
Is it bad to have it, even if unused in code (only tested)?

Reply:
Good question.  The biggest mistake I made in doing this OAUTHBEARER implementation was trying to do too much.  The community reigned me in over time :-)  My guess would be to keep things as simple and minimalistic as possible; there is less that can go wrong, less to debate over/review/test/fix, and ultimately a shorter time to when the code actually gets merged.  At least that's the lesson I took from the experience.

Reply:
That's true in general. I honestly believe this is a useful constructor to have though